### PR TITLE
Support reporting errors in afterAll.

### LINKED ISSUE #210: Errors in afterAll functions are not reported.
## Feature request 

**Current behavior**
Errors in afterAll() still allow tests to pass.
**Wanted behavior**
Tests fail on errors
**What is the motivation / use case for changing the behavior?**
Errors are bad ;-)

The suiteDone() can take a result argument:
https://github.com/bcaudan/jasmine-spec-reporter/blob/master/src/display/execution-display.ts#L101
jasmine will report afterAll errors as a failedExpectation in that arg.

We need that to 1) report the error 2) ensure the final status is FAILED.

### PR DESCRIPTION
**Pull request by johnjbarton**
Starting in jasmine 2.5, errors in afterAll are reported to jasmineDone or suiteDone.
Add ExecutionMetrics.errors to store these errors; report them.

Fixes #210

### CODE REVIEW

File: src/execution-metrics.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export class ExecutionMetrics {
     public skippedSpecs = 0;
     public totalSpecsDefined = 0;
     public executedSpecs = 0;
+    public errors: CustomReporterResult[] = [];
```
Comment (bcaudan) [inline src/execution-metrics.ts]: In jasmine default reporter, this kind of error is specific to afterAll error, what about naming it `afterAllErrors` instead of `errors`?
Then, the summary report should also be something like `* AfterAll Errors *`

File: src/display/summary-display.ts [hunk 2]
```diff
@@ -103,4 +109,32 @@ export class SummaryDisplay {
...
+            this.errorSummary(errors[i], i + 1);
+            this.logger.newLine();
+        }
+        this.logger.newLine();
+        this.logger.resetIndent();
+    }
+
+    private errorSummary(error: CustomReporterResult, index: number): void {
+        this.logger.log(`${index}) ${error.fullName}`);
+        if (this.configuration.summary.displayErrorMessages) {
```
Comment (bcaudan) [inline src/display/summary-display.ts]: These errors are not related to failed assertion, I would be in favor of always displayed it for now

Comment (johnjbarton) [inline src/display/summary-display.ts]: done

File: src/execution-metrics.ts (hunk 1 above)
Comment (johnjbarton) [inline src/execution-metrics.ts]: I was planning to cheat and use the same mechanism in karma-jasmine, for global errors.

In karma-jasmine, global errors have to be passed from the browser back to the server, then reported via the karma-reporter. Our reporter translates karma API into jasmine calls then into jasmine-spec-reporter.  The `jasmineDone()` api provides a path to report global errors.

That case need not concern you, so my only other question is whether there could be other errors in future? The `jasmineDone()`/`suiteDone()` reporting path is not specific to `afterAll()`, but then again I don't know of any other possible errors.

Not a big deal, so decide and I will follow up.

**Comment by johnjbarton** [comment]
For a case like
```js
describe('an error in describe callback', () => {
	it('should not pass', () => {
		expect(true).toBe(true);
	});
	throw new Error('describe error');
});
```
We see:
```shell
Jasmine started

  an error in describe callback
    ✓ should not pass
    ✗ encountered a declaration exception
      - Error: describe error

**************************************************
*                    Failures                    *
**************************************************

1) an error in describe callback encountered a declaration exception
  - Error: describe error

Executed 2 of 2 specs (1 FAILED) in 0.011 sec.
```
So jasmine chose to create a new spec and attach the error to it.  Does that change your thinking? 

Not a big deal to me, but I do think that nesting (indenting) the suite error under the suite makes sense, but it could be w/o the x for example.

(I will follow up in one week after my vacation).

File: src/execution-metrics.ts (hunk 1 above)
Comment (bcaudan) [inline src/execution-metrics.ts]: I'm ok with global errors

**Comment by bcaudan** [comment]
I was not aware of this behavior. 
I am ok to keep the display as is for now.

File: src/execution-metrics.ts (hunk 1 above)
Comment (johnjbarton) [inline src/execution-metrics.ts]: done

**Comment by johnjbarton** [comment]
I think I addressed your concerns, if not let me know.
